operator/pkg/ciliumendpointslice/cescache.go

```diff
@@ -94,6 +97,82 @@ func newCESCache() *CESCache {
 	}
 }
 
+// Add CEP to cache, map to CES name and node name
+func (c *CESCache) addCEP(cepName CEPName, cesName CESName, nodeName NodeName, gidLabels Labels) {
+	c.updateCEPInCache(cepName, nodeName, gidLabels, cesName)
+}
+
+// Add or update CEP in cache, map to CES name, node name and associated CID
+func (c *CESCache) upsertCEP(cepName CEPName, cesName CESName, nodeName NodeName, gidLabels Labels, cid CID) {
+	c.updateCEPInCache(cepName, nodeName, gidLabels, cesName)
+	c.globalIdLabelsToCIDSet[gidLabels].setSelectedID(cid)
+	c.globalIdLabelsToCIDSet[gidLabels].ids.Insert(cid)
+	c.cidToGidLabels[cid] = gidLabels
```

Comment:
This doesn't look correct to me, because it manually updates the CID/label mappings, bypassing the  `insertCID/deleteCID` methods. Which in turn can lead to stale or out-of-sync state, as the invariants are no longer guaranteed (e.g., in case this method gets called after that the deletion even of a CID has already been seen). I'd suggest to simplify the logic by always looking into the CESCache to retrieve the identity to CID mapping, and simply trigger a retry if not available yet.